### DISCUSSION TITLE
datasets/explorer/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8044] [Docs]
# datasets/explorer/

Discover the Ultralytics Explorer, a versatile tool and Python API for CV dataset exploration, enabling semantic search, SQL queries, and vector similarity searches.

https://docs.ultralytics.com/datasets/explorer/

**Comment by AloshkaD** [comment]
The yolo explorer command will open the explorer on the local machine. How would I set it up to work on AWS while I'm connected with my local machine using ssh?

**Reply by pderrenger** [reply]
To set up the Ultralytics Explorer to work on an AWS instance while you're connected via SSH from your local machine, you can use port forwarding to access the Explorer's web interface. Here's a concise step-by-step guide:

1. **SSH into AWS with Port Forwarding:**
   Use the `-L` option to forward a local port to a port on the AWS instance. Replace `local_port` with a port on your local machine (e.g., `8080`), and `aws_port` with the port the Explorer runs on (usually `5000`).

   ```bash
   ssh -i /path/to/your/key.pem -L local_port:localhost:aws_port user @AloshkaD
   ```

2. **Run Explorer on AWS:**
   Once connected, start the Explorer on the AWS instance:

   ```bash
   yolo explorer
   ```

3. **Access Explorer Locally:**
   Open a web browser on your local machine and navigate to `localhost:local_port` to access the Explorer interface.

Remember to replace `/path/to/your/key.pem` with the path to your AWS key, `user` with your AWS username, and `aws_instance_ip` with your AWS instance's IP address. Also, ensure that the AWS security group allows inbound traffic on the `aws_port`.

**Reply by AloshkaD** [reply]
Worked with a minor tweak to reference my config file. Thank you!

**Reply by pderrenger** [reply]
@AloshkaD you're welcome! I'm glad to hear that you were able to get it working with your config file. If you have any more questions or need further assistance in the future, don't hesitate to reach out. Happy exploring with Ultralytics Explorer! 🚀😊

**Comment by BABAFAREEDDUDEKULA** [comment]
how can i run above mentioned AI using yolo v8
previously i have tried yolo v9,v8,v7,rt-detr

**Reply by AloshkaD** [reply]
Can you be more specific?

**Reply by glenn-jocher** [reply]
Hey there! 🌟

It sounds like you've got a pretty wide range of experience with various YOLO versions and even RT-DETR! For running AI using YOLOv8 specifically, the process is smoothly integrated into the Ultralytics framework, much like the versions you've mentioned.

To get started, make sure you have Ultralytics YOLOv8 installed. If not, you can do so via pip:

```bash
pip install ultralytics
```

And then, to run your AI model with YOLOv8, you can simply use:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')  # Here 'yolov8n.pt' is an example model. Replace it with your model path.

# Run inference
results = model('path_to_your_image_or_video')
```

Or, if you prefer using the command line:

```bash
yolo detect predict model=yolov8n.pt source=path_to_your_image_or_video
```

Remember to replace 'yolov8n.pt' and 'path_to_your_image_or_video' with the actual paths to your YOLOv8 model and the image or video you want to run inference on. 🚀

Since you've used various versions before, the process should feel quite familiar. If there's more specific functionality from the Ultralytics Explorer that you're interested in integrating with YOLOv8, such as semantic search or SQL queries, the installation command provided in the documentation will be your first step:

```bash
pip install ultralytics[explorer]
```

After that, you can launch the Ultralytics Explorer with:

```bash
yolo explorer
```

And dive into explore mode! For any further details or if you've got a unique use case in mind, feel free to share more specifics. Happy to help further. Happy coding! 🎉

**Reply by BABAFAREEDDUDEKULA** [reply]
Thanks a lot my brother

**Reply by glenn-jocher** [reply]
@BABAFAREEDDUDEKULA hey there! 😄

Thanks for reaching out. If you're interested in diving into the Ultralytics Explorer to explore CV datasets with semantic search, SQL queries, vector similarity search, and more, it sounds like you're on the right track!

To get started, ensure you've got the Explorer installed via:

```bash
pip install ultralytics[explorer]
```

Once installed, you can jump straight into using the GUI by typing:

```bash
yolo explorer
```

This launches the GUI in your browser, where you can experiment with creating embeddings, conducting image searches, and even using SQL or natural language queries!

If you encounter any issues or have specific questions as you use the Ultralytics Explorer, feel free to drop them here. Happy exploring! 🌟

**Comment by mkutu** [comment]
how can I set this up with a custom dataset? Could you direct me to any resources. Thanks!

**Reply by glenn-jocher** [reply]
@mkutu hello!

Setting up YOLOv8 with a custom dataset can be straightforward! You'd primarily need a data YAML file to describe your dataset structure and then use it in your training command. Here’s a basic guide to get you started:

1. **Prepare your data**: Organize your dataset into two folders (images and labels), with corresponding files named identically.
2. **Create a YAML file**: This file should specify the paths to your training and validation data, the number of classes, and class names. Here's an example:

   ```yaml
   # custom_data.yaml
   train: path/to/train/images
   val: path/to/val/images
   nc: number_of_classes
   names: ['class1', 'class2', 'class3']
   ```

3. **Start Training**: Use the following command to train your model:
   ```bash
   yolo train data=custom_data.yaml model=yolov8n.yaml
   ```

For exploring and managing your datasets efficiently, you might consider using the [Ultralytics Explorer](https://docs.ultralytics.com/datasets/explorer/). This tool offers features like semantic search, SQL queries, and even AI-based querying to derive insights from your data.

Hope this helps! If you have any more questions, feel free to ask.

**Comment by pooya-github** [comment]
Hi,

In colab code after running !pip install ultralytics[explorer] and running: from ultralytics import explorer i got the following error:

ImportError                               Traceback (most recent call last)
<ipython-input-14-598562853d02> in <cell line: 1>()
----> 1 from ultralytics import explorer

ImportError: cannot import name 'explorer' from 'ultralytics' (/usr/local/lib/python3.10/dist-packages/ultralytics/__init__.py)

how can i fix it?

**Reply by glenn-jocher** [reply]
@pooya-github the `Ultralytics Explorer` feature has been deprecated as of `ultralytics>=8.3.10`. To access similar or enhanced functionality, we recommend using [Ultralytics HUB](https://hub.ultralytics.com/), our no-code platform for dataset exploration and management. Let me know if you have further questions!
